misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormattingTreeModifier.java

```diff
@@ -1631,13 +1634,9 @@ public MarkdownDocumentationLineNode transform(MarkdownDocumentationLineNode mar
         Token hashToken;
 
         if (markdownDocumentationLineNode.documentElements().isEmpty()) {
-            hashToken = formatToken(markdownDocumentationLineNode.hashToken(), env.trailingWS, env.trailingNL);
+            hashToken = formatToken(markdownDocumentationLineNode.hashToken(), 0, 1);
```

Comment:
In the formatter implementation, we take the last element's trailing whitespaces and the trailing newlines from the parent, therefore, it would be more appropriate if we could go ahead with the same format as follows. This might need to be fixed in the other related nodes as well.
```suggestion
            hashToken = formatToken(markdownDocumentationLineNode.hashToken(), env.trailingWS, env.trailingNL);
```